backend/session.go

```diff
@@ -476,7 +476,15 @@ func (s *session) handleSendCommand(cmd *proto.SendCommand) (interface{}, error)
 		return nil, err
 	}
 
-	// TODO: verify parent
+	isValidParent, err := s.room.IsValidParent(cmd.Parent)
+	if err != nil {
+		return nil, err
+	}
+	if !isValidParent {
+		if err := s.sendBounce("invalid parent ID"); err != nil {
```

Comment:
In heim terms, "bounce" means "disconnect from the server," not message rejection. It would be more appropriate here to just return a meaningful error (which should be added to proto/errors.go, something like ErrMessageParentNotFound). An error return will be translated into an error-reply, sent back to the client.